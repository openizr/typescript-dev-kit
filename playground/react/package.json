{
  "license": "MIT",
  "devDependencies": {
    "@types/prop-types": "^15.7.4",
    "@types/react": "^17.0.26",
    "@types/react-dom": "^17.0.9",
    "typescript-dev-kit": "file:../../../../typescript-dev-kit/dist"
  },
  "dependencies": {
    "basx": "^1.3.7",
    "diox": "^4.0.6",
    "prop-types": "^15.7.2",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "eslintConfig": {
    "extends": [
      "./node_modules/typescript-dev-kit/main.js"
    ]
  },
  "tsDevKitConfig": {
    "target": "web",
    "devServer": {
      "ip": "0.0.0.0",
      "port": 5001
    },
    "html": {
      "template": "./html/index.html",
      "entries": [
        "sass",
        "typescript"
      ]
    },
    "entry": {
      "sass": "./scripts/sass.ts",
      "typescript": "./scripts/typescript.tsx",
      "javascript": "./scripts/javascript.jsx"
    },
    "splitChunks": true,
    "srcPath": "src",
    "distPath": "public",
    "banner": "/*!\n * Copyright (c) Matthieu Jabbour.\n * All rights reserved.\n */",
    "env": {
      "development": {
        "ENV": "development",
        "NODE_ENV": "development"
      },
      "production": {
        "ENV": "production",
        "NODE_ENV": "production"
      }
    }
  },
  "scripts": {
    "init": "node_modules/typescript-dev-kit/scripts/init.sh",
    "dev": "cd node_modules/typescript-dev-kit/ && node scripts/dev",
    "build": "cd node_modules/typescript-dev-kit/ && node scripts/build && cd ../../ && sed -i -e \"s%/\\*WEBPACK_CSS\\*/%WEBPACK_CSS\\n%g\" public/index.html  && sed -i -e \"/WEBPACK_CSS/r $(ls public/assets/styles/common.*.css)\" public/index.html && sed -i -e \"s/WEBPACK_CSS//g\" public/index.html",
    "test": "cd node_modules/typescript-dev-kit/ && node scripts/test",
    "doc": "node_modules/.bin/typedoc --out ./docs/ --exclude \"**/__+(tests|mocks)__/**\" src/scripts/javascript.jsx",
    "postinstall": "rm -f node_modules/.eslintcache"
  }
}
