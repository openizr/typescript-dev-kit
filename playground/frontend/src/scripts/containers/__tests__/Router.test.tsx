/**
 * @jest-environment jsdom
 */

import * as React from 'react';
import store from 'scripts/store/index';
import Router from 'scripts/containers/Router';
import { render } from '@testing-library/react';

type Misc = any; // eslint-disable-line @typescript-eslint/no-explicit-any

// Useful mocks allowing us to easily test React lazy components and Suspense.
jest.mock('react', () => {
  const MockedReact = jest.requireActual('react');
  MockedReact.Suspense = ({ children, fallback }: Misc): Misc => (
    process.env.LOADING === 'true' ? fallback : children
  );
  MockedReact.lazy = (callback: Misc): Misc => callback();
  return MockedReact;
});

jest.mock('scripts/store/routes', () => ({
  // eslint-disable-next-line global-require, @typescript-eslint/no-var-requires
  '/': (): Misc => require('scripts/pages/Home').default,
  // eslint-disable-next-line global-require, @typescript-eslint/no-var-requires
  '/js': (): Misc => require('scripts/pages/HomeJS').default,
}));

describe('react/Router', () => {
  beforeEach(() => {
    process.env.LOADING = 'false';
    jest.clearAllMocks();
  });

  test('renders correctly - loading page', () => {
    process.env.LOADING = 'true';
    const { container } = render(<Router locale={{ LABEL_TEST: 'Test' }} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - found page', () => {
    store.mutate('router', 'NAVIGATE', '/');
    const { container } = render(<Router locale={{ LABEL_TEST: 'Test' }} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - not found page', () => {
    store.mutate('router', 'NAVIGATE', '/404');
    const { container } = render(<Router locale={{ LABEL_TEST: 'Test' }} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
