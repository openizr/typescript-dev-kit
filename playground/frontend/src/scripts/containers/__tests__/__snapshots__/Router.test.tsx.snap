// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react/Router renders correctly - found page 1`] = `
<p>
  <img
    alt="test"
    src="file"
  />
  <img
    alt="test"
    src="file"
  />
  Test
</p>
`;

exports[`react/Router renders correctly - loading page 1`] = `
<div>
  Loading...
</div>
`;

exports[`react/Router renders correctly - not found page 1`] = `null`;
